packages/jellyfish-transaction/src/script/dftx/dftx_icxorderbook.ts

```diff
@@ -31,3 +32,47 @@ export class CICXSubmitDFCHTLC extends ComposableBuffer<ICXSubmitDFCHTLC> {
     ]
   }
 }
+
+/**
+ * ICXSubmitEXTHTLC DeFi transaction
+ */
+export interface ICXSubmitEXTHTLC {
+  offerTx: string // -----------| 32 byte, txid for which offer is this HTLC
+  amount: BigNumber // ---------| 8 byte, amount that is put in HTLC
+  hash: string // --------------| 32 byte, hash for the hash lock part
+  htlcScriptAddress: string // -| n = VarUInt{1-9 bytes}, + n bytes, script address of external htlc
+  ownerPubkey: string // -------| n = VarUInt{1-9 bytes}, + n bytes, pubkey of the owner to which the funds are refunded if HTLC timeouts
+  timeout: number // -----------| 4 byte, timeout (absolute in block) for expiration of external htlc in external chain blocks
+}
+
+/**
+ * Composable ICXSubmitEXTHTLC, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+export class CICXSubmitEXTHTLC extends ComposableBuffer<ICXSubmitEXTHTLC> {
+  static OP_CODE = 0x34
+  static OP_NAME = 'OP_DEFI_TX_ICX_SUBMIT_EXT_HTLC'
+
+  composers (msg: ICXSubmitEXTHTLC): BufferComposer[] {
+    return [
+      ComposableBuffer.hexBEBufferLE(32, () => msg.offerTx, v => msg.offerTx = v),
+      ComposableBuffer.satoshiAsBigNumber(() => msg.amount, v => msg.amount = v),
+      ComposableBuffer.hexBEBufferLE(32, () => msg.hash, v => msg.hash = v),
+      ComposableBuffer.varUIntUtf8BE(() => msg.htlcScriptAddress, v => msg.htlcScriptAddress = v),
+      { // NOTE(surangap): may be use optionalVarUIntHex when available or move this piece of code to buffer_composer.ts
```

Comment:
In that case it might be better to add VarUIntHex to buffer_composer.ts and test it independently as ownerPubkey is not optional